packages/vscode-extension/src/debug/teamsfxTaskHandler.ts

```diff
@@ -64,6 +75,20 @@ function onDidEndTaskProcessHandler(event: vscode.TaskProcessEndEvent): void {
   const task = event.execution.task;
   if (task.scope !== undefined && isTeamsfxTask(task)) {
     allRunningTeamsfxTasks.delete({ source: task.source, name: task.name, scope: task.scope });
+  } else if (isNpmInstallTask(task)) {
+    activeNpmInstallTasks.delete(task.name);
+    // when the task in the active terminal is ended.
+    // TODO: only when the task is ended successfully.
+    if (vscode.window.activeTerminal?.name === task.name) {
+      for (const hiddenTaskName of activeNpmInstallTasks) {
```

Comment:
If a hidden task is not found in ternimals, should it be removed from active task list?

Reply:
it can be safely kept in the active task list as it's a real hidden npm-install task, and it won't break the logic here.

Reply:
But it may impact the next run. The active task list is never cleaned until user closes vscode...

Reply:
Offline discussed that the active task list might be impact only when the task.json is modified by user manually.